spec/models/user_spec.rb

```diff
@@ -87,13 +87,62 @@
   end
 
   describe 'methods' do
+    describe '#attended_event?' do
+      context 'user has attended to the event' do
+        before do
+          registration = create(:registration, user: user, conference: conference)
+          create(:events_registration, event: event1, registration: registration, attended: true)
+        end
+
+        it 'returns true' do
+          expect(user.attended_event?(event1)).to be true
+        end
+      end
+
+      context 'user did not register for the event' do
+        it 'returns false' do
+          expect(user.attended_event?(event1)).to be false
+        end
+      end
+
+      context 'user registered for the event, but did not attend' do
+        before do
+          registration = create(:registration, user: user, conference: conference)
+          create(:events_registration, event: event1, registration: registration)
+        end
+
+        it 'returns false' do
+          expect(user.attended_event?(event1)).to be false
+        end
+      end
+    end
+
     describe '#name' do
       it 'returns the username as name if there is not name' do
         user = create(:user, name: nil)
         expect(user.name).to eq(user.username)
       end
     end
 
+    describe '#registered_to_event?' do
+      context 'user has registered to event' do
+        before do
+          registration = create(:registration, user: user, conference: conference)
```

Comment:
do this in a let, then it is created when needed, can be reused and then you have all the object created for your tests together and at the beginning. 

And as you are using it for both `#attended_event?` and `#registered_to_event?` you can move it up (just after `describe 'methods' do`) so you only need to define it once. :slightly_smiling_face: